controllers/metal3cluster_controller.go

```diff
@@ -97,6 +97,7 @@ func (r *Metal3ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		error := capierrors.InvalidConfigurationClusterError
 		metal3Cluster.Status.FailureReason = &error
 		metal3Cluster.Status.FailureMessage = pointer.StringPtr("Unable to get owner cluster")
+		conditions.MarkFalse(metal3Cluster, capm3.BaremetalInfrastructureReadyCondition, capm3.ControlPlaneEndpointFailedReason, capi.ConditionSeverityError, err.Error())
```

Comment:
Is it correct that we are telling that infra is not ready due to `ControlPlaneEndpointFailedReason` while it is due to unavailability of fetching the cluster?

Reply:
Thank you for catching that.
I replaced this with a generic `InternalFailureReason` to capture such failures. I don't think every minor failure or error case, needs a Condition.